pkg/controller/scale_up.go

```diff
@@ -62,17 +44,37 @@ func (c *Controller) ScaleUp(opts scaleOpts) (int, error) {
 	return untainted, nil
 }
 
+// Calulates how many new nodes need to be created
+func (c *Controller) calculateNodesToAdd (nodesToAdd int64, TargetSize int64, MaxNodes int64) (int64) {
+	// Clamp it to the max if exceeding max target size
+	if TargetSize + nodesToAdd > MaxNodes {
+		nodesToAdd = MaxNodes - TargetSize
+		log.Infof("increasing nodes exceeds maximum (%v). Clamping add amount to (%v)", MaxNodes, nodesToAdd)
+	}
+	return nodesToAdd
+}
+
 // scaleUpCloudProviderNodeGroup increases the size of the cloud provider node group by opts.nodesDelta
 func (c *Controller) scaleUpCloudProviderNodeGroup(opts scaleOpts) (int, error) {
-	nodegroupName := opts.nodeGroup.Opts.Name
-	nodesToAdd := int64(opts.nodesDelta)
 
 	cloudProviderNodeGroup, ok := c.cloudProvider.GetNodeGroup(opts.nodeGroup.Opts.CloudProviderGroupName)
 	if !ok {
 		return 0, fmt.Errorf("cloud provider node group does not exist: %s", opts.nodeGroup.Opts.CloudProviderGroupName)
 	}
 
-	if nodesToAdd+cloudProviderNodeGroup.TargetSize() <= cloudProviderNodeGroup.MaxSize() {
+	nodegroupName := opts.nodeGroup.Opts.Name
+	nodesToAdd := c.calculateNodesToAdd(int64(opts.nodesDelta), cloudProviderNodeGroup.TargetSize(), cloudProviderNodeGroup.MaxSize())
+	if nodesToAdd <= 0 {
+		err := fmt.Errorf(
+			"the number of nodes(%v) is more than specified maximum of %v. Taking no action",
```

Comment:
Can you update the error message to be more clear as to what the issue is - that we have reached the max size?